data/transactions/verify/txn_test.go

```diff
@@ -962,3 +962,15 @@ func BenchmarkTxn(b *testing.B) {
 	}
 	b.StopTimer()
 }
+
+func TestReturnUnverified(t *testing.T) {
```

Comment:
Note: the tests will be repositioned as part of phase 2. Leaving them in streamverifier_test.go in this PR to make the code review diffs easier to follow. 

https://app.zenhub.com/workspaces/algorand-main-workspace-5ea71fda6d554d53dda36f16/issues/gh/algorand/go-algorand-internal/2778